### DISCUSSION TITLE
Why does a Filament notification appear 3 times at once?

### DISCUSSION THREAD

**Discussion by MurDev** [filamentphp/filament#5883] [Help]
I have a table that displays products and I've created a custom action that allows users to add products to a shopping cart. However, I want to perform a verification and send a notification. In the class of my Action, when I process the method url, the notification appears multiple times as soon as I load the product page.

There is the code of my action class :
`<?php

namespace Filament\Tables\Actions;

use Closure;
use Filament\Notifications\Notification;
class AddToCartAction extends Action
{

    protected ?Closure $mutateRecordDataUsing = null;

    public static function getDefaultName(): ?string
    {
        return 'add';
    }

    protected function setUp(): void
    {
        parent::setUp();

        $this->label(__('Add'));

        $this->modalHeading(__('Add product item to cart'));

        $this->url(
            function ($record){
                if($record->stock == 0){
                    Notification::make()
                    ->warning()
                    ->title('Out of stock')
                    ->send();
                }else{
                    route('filament.pages.product-to-add', ['record' => $record]);
                }
            }
        );

        $this->modalActions(fn (): array => array_merge(
            $this->getExtraModalActions(),
            [$this->getModalCancelAction()->label(__('filament-support::actions/view.single.modal.actions.close.label'))],
        ));

        $this->color('secondary');

        $this->icon('heroicon-s-plus');

        $this->successNotificationTitle(__('Added to the cart'));

        $this->action(static function (): void {
        });
    }

> 

> }`

**Comment by danharrin** [comment]
The URL closure might be evaluated more than once, so maybe instead of passing a URL, pass an action() and redirect to the URL
